q2_quality_control/_threshold_graph/_visualizer.py

```diff
@@ -60,31 +95,112 @@ def decontam_score_viz(output_dir, decontam_scores: qiime2.Metadata,
     gray_lab_arr = []
     red_lab_arr = []
     blue_lab_arr = []
+    data_arr = []
+    display_seqs_arr = []
+    true_fasta_dest = []
+    contam_fasta_dest = []
+    nan_fasta_dest = []
+    sorted_key_arr = []
+    feature_or_read_arr = []
 
+    #iterates through tables and keys of ASV tables and decontam score tables
     for key in table_dict.keys():
         table = table_dict[key]
         decontam_scores = decontam_scores_dict[key]
-        df = decontam_scores.to_dataframe()
-        if df['p'].isna().all():
+        if decontam_scores['p'].isna().all():
             raise ValueError("No p-values exist for the data provided.")
 
+        #pull out relevant data from objs
         read_nums = table.sum(axis='rows')
-
-        p_vals = df['p'].dropna()
+        p_vals = decontam_scores['p'].dropna()
         filt_read_nums = read_nums[p_vals.index]
 
+        #start ASV contaminant differentiation
         contams = (p_vals < threshold)
+        contam_indices = []
+        true_indices = []
+
+        #parses index for contaminant identification
+        for true_or_false_index in contams.index:
```

Comment:
See my previous comment - you shouldn't need to do all of this.

Reply:
This was removed in favor of dataframe slicing

addressed in [adfe12a](https://github.com/qiime2/q2-quality-control/pull/82/commits/adfe12ad4213c9923face4267028d18902427b8a)